### PR TITLE
Added prompt to create account before upvoting questions on the Readi…

### PR DESCRIPTION
**Pull request by msl5994**
…ngPage and QuestionDetails page.

### CODE REVIEW

**Review by vteague — CHANGES_REQUESTED** [review]
I think it might have been a mistake to remove the
        if (!AlreadyUpvoted)
condition at line 137 of Question.cs. Or possibly we need an equivalent condition when we're un-up-voting things, to make sure you can only un-up-vote if(AlreadyUpvoted).

As you can see, I managed to create a situation where there are -1 votes:
Screenshot_1651116408

I did this by upvoting on the reading page, then tapping on the question to go to the question detail page, then tapping the voting button to un-up-vote it, then going back to the reading page and tapping on the (still green) upvote button to un-up-vote it.  (Clearly the upvote status on the reading page is partly working, because the number of upvotes had correctly gone down to zero before I tapped, but the upvote icon remained green (wrong) even though I had un-up-voted it in the QuestionDetailPage)

In short, I think that both the upvote button on the ReadingPage and the one on the QuestionDetailPage need to check that you're only allowed to upvote something that you're not currently in the AlreadyUpvoted state for, and conversely that you can't un-up-vote something that you _are_ currently in the AlreadyUpvoted state for.

I am not certain whether this used to be correct, and these changes have introduced this bug, or whether it was never exactly right and we've only just noticed now, but either way worth fixing before merge.

**Comment by vteague** [comment]
Otherwise the actual purpose of the branch, to require people to register before upvoting, looks great and all seems to be working.

**Comment by vteague** [comment]
This is all good, thanks, must have been looking at an old version when I got that weird behaviour. Now looking perfect.
